src/GraphQL.AotCompilationSample.TypeFirst/Program.cs

```diff
@@ -0,0 +1,103 @@
+Console.WriteLine();
+
+IServiceCollection serviceCollection = new ServiceCollection();
+// build services as usual; most methods are fully supported except:
+//   - AddClrTypeMappings
+//   - AddAutoClrMappings
+//   - AddAutoSchema
+serviceCollection.AddGraphQL(b => b
+    .AddSystemTextJson()
+    .AddAutoSchema<StarWarsQuery>(c => c.WithMutation<StarWarsMutation>())
+);
+
+#pragma warning disable IL2111 // Method with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
+// all types within GraphQL.StarWars.TypeFirst are rooted via TrimmerRootAssembly in the csproj so they will not be trimmed
+// for enumeration types, must also root these two types (for each enum in the schema)
+Preserve<GraphQLClrOutputTypeReference<Episodes>>();
+Preserve<EnumerationGraphType<Episodes>>();
+// for connection types, must also root the connection types used in the schema
+Preserve<Connection<IStarWarsCharacter>>();
+Preserve<Edge<IStarWarsCharacter>>();
+Preserve<PageInfo>();
+#pragma warning restore IL2111 // Method with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
+
+serviceCollection.AddSingleton<StarWarsData>();
+
+// must manually register the query and mutation types or AOT will trim their constructors
+// all other graph types' constructors are preserved via calls to Field<T>
+//serviceCollection.AddTransient<StarWarsQuery>();
+//serviceCollection.AddTransient<StarWarsMutation>();
+
+// other notes:
+// - auto clr type mappings are generally not supported
+// - auto registering graph types are generally not supported
+// - field builders that use Expressions as resolvers such as Field(x => x.Name) are discouraged
+// - field builders that do not include a resolver such as Field<StringGraphType>("Name") are not supported
+// - strongly recommend each field has the explicit graph type specified and a resolver specified
+
+#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+var services = serviceCollection.BuildServiceProvider();
+#pragma warning restore IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+
+var executer = services.GetRequiredService<IDocumentExecuter>();
+
+var ret = await executer.ExecuteAsync(new ExecutionOptions
+{
+    Schema = services.GetRequiredService<ISchema>(),
+    Query = "{ hero { id name } }",
+    RequestServices = services,
+    ThrowOnUnhandledException = true,
+}).ConfigureAwait(false);
+
+var serializer = services.GetRequiredService<IGraphQLTextSerializer>();
+var response = serializer.Serialize(ret);
+
+Console.WriteLine("Executing request: { hero { id name } }");
+Console.WriteLine();
```

Comment:
I suggest to move two lines before `var ret = ...`.

Reply:
or before `var executer = `

Reply:
Done